### PR TITLE
[RF] Update reference files for testProxiesAndCategories

### PR DESCRIPTION
**Pull request by guitargeek**
* update the categories_v622.root file to actually use the class
  versionse from ROOT 6.22

* add new categories_v624.root reference file and use it in RooCategory
  I/O unit tests

### CODE REVIEW

**Review by hageboeck — CHANGES_REQUESTED** [review]
You need to add a `COPY_FILES` or similar to the `CMakeLists.txt` where this test is defined. Otherwise, it won't find the file.

**Comment by guitargeek** [comment]
Ah yes, right! I have changed it now.

File: roofit/roofitcore/test/CMakeLists.txt [hunk 1]
```diff
@@ -8,7 +8,7 @@
 # @author Patrick Bos, NL eScience Center, 2018
 
 ROOT_ADD_GTEST(simple simple.cxx LIBRARIES RooFitCore)
-ROOT_ADD_GTEST(testRooCacheManager testRooCacheManager.cxx LIBRARIES RooFitCore)
+ROOT_ADD_GTEST(testRooCacheManager testRooCacheManager.cxx LIBRARIES RooFitCore RooFit)
```
Comment (hageboeck) [inline roofit/roofitcore/test/CMakeLists.txt:11]: Was this change intentional?

**Review by hageboeck — APPROVED** [review]
👍
